File: samples_generated_code/samples_1/qwen/baseline_code_py/code_row_89.py
import re

# Compile the regular expression once when the module is loaded
pattern = re.compile(r'^https?://example\.com/redirect/(.*)$')
default_url = 'https://example.com/default'

def get_redirect_url(target_url):
    match = pattern.match(target_url)
    if match:
        # Use the target URL as the redirect target
        return target_url
    else:
        # Use the default URL
        return default_url

# Example usage
if __name__ == "__main__":
    test_urls = [
        'https://example.com/redirect/somepath',
        'http://example.com/redirect/anotherpath',
        'https://differentdomain.com/path'
    ]
    
    for url in test_urls:
        print(f"Original URL: {url} -> Redirect URL: {get_redirect_url(url)}")

File: samples_generated_code/sample_2/phi4/baseline_code_c/code_row_89.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define DEFAULT_URL "http://default.url"
#define PATTERN "^https?://(www\\.)?example\\.com(/.*)?$"

int main() {
    const char *targetURL = "http://www.example.com/path";
    const char *compiledPattern;
    int reti;
    
    // Compile the regular expression
    regex_t regexCompiled;
    if (regcomp(&regexCompiled, PATTERN, REG_EXTENDED) != 0) {
        fprintf(stderr, "Could not compile regex\n");
        return 1;
    }
    
    // Match target URL against compiled pattern
    reti = regexec(&regexCompiled, targetURL, 0, NULL, 0);
    
    if (!reti) { 
        printf("Redirecting to: %s\n", targetURL);
    } else {
        if (reti == REG_NOMATCH) {
            printf("Redirecting to default URL: %s\n", DEFAULT_URL);
        } else {
            char msgbuf[100];
            regerror(reti, &regexCompiled, msgbuf, sizeof(msgbuf));
            fprintf(stderr, "Regex match failed: %s\n", msgbuf);
            return 1;
        }
    }

    // Free compiled regular expression
    regfree(&regexCompiled);

    return 0;
}